### PR TITLE
Qt5: make transport icons appear, fix some warnings

### PR DESCRIPTION
**Pull request by edogawa23**
Hello, a while ago I found your Qt5 port branch (by chance). I'd love to see reZound alive and think Qt UI is the way forward, regarding portability, hidpi displays andin general :) Now since the recent addition of telemetry to audacity, I observed a lot of interest for alternatives in the Linux audio community.

So I tried to compile your branch, and it worked right away except for minor glitches like not showing the transport button icons, and tried my best to fix the couple of deprecation warnings...

I have only started to test the application and so far haven't experienced a crash or any bad behaviour, in fact it seems to be very fast and stable. I see that there are large parts of Qt code inside "#if 0" statements, but haven't looked in detail.

So I want to ask what your future plans are for reZound, and comment on this PR, as I'm all but an experienced coder...

### CODE REVIEW

File: src/qt/ProgressDialog.h [hunk 1]
```diff
@@ -14,15 +14,14 @@ class ProgressDialog : public QDialog, private Ui::ProgressDialog
 	Q_OBJECT
 
 public:
-    ProgressDialog(QWidget * parent , const string title,bool showCancelButton) :
```
Comment (ddurham2) [inline src/qt/ProgressDialog.h:17]: Why remove title and cancel button?

**Comment by ddurham2** [comment]
Hey, thanks for the contribution!  I just have that one question about it.

Yeah, honestly it's just not a project that I've made a lot of time for.  I use ReZound several times a week, but I'm using the libfox ui which is "good enough" for me.  I started the Qt port way back around the time that debian stopped packaging libfox.  I figured porting to Qt would allow it to more widely available on distroes.  I use Qt (widgets, not so much QML) professionally and so it comes pretty natural, but I just haven't made the time to continue work.

Also, I ditched autotools for cmake, which I think is much better now.

I should though.  There are a few crashes and/or race conditions here and there (esp when you edit while playing).  They don't happen often, but it's something that needs fixing. I should probably run with valgrind's helgrind to see what it says.

Also TPoolFile (which is the "virtual memory" layer for the working file) has some edge conditions somewhere.  Occasionally, it will barf.  I figure some thorough unit tests (w/ e.g. google-test) could find those problems.

**UI:**
The UI is old school obviously, and needs some updating.  The synchronous nature of the UX and modal dialogs used throughout can be annoying at times.

All images need to be recreated as svg.  Qt's happen to handle them.

I've never loved the file listing at the top.  It doesn't scale well when having many files open.  A tabbed view (MDI) might be more appropriate.

The status area below the waveform view could be improved too.

Various other things.

Anyway... all that's beyond just catching up and having parity with the fox UI.  I presently don't remember exactly where I left off on the Qt effort.

I could probably be persuaded to pick things back up.  It's just that I have several other personal projects going on.  One of them is an audio project which is part of the reason I still use ReZound as often as I do.

File: src/qt/ProgressDialog.h (hunk 1 above)
Comment (edogawa23) [reply src/qt/ProgressDialog.h:17]: uhm, tbh i don't quite remember, i may have been confused by a warning or copile errors that i cannot even reproduce anymore, after reverting this.

Sorry for that.

**Comment by edogawa23** [comment]
Hi Davy,

thanks for responding, I'd like to persuade you to pick things back up ;-)

I'm reading the Qt code and have some difficulties following it...

For example I tried to load the transport icons programatically via transportWindow constructor, and couldn't make it work, so I assigned them in Designer, directly in the .ui file... That doesn't work for the Menu bar as that is created dynamically in code too, and no matter what I tried (spent a few hours just on this yesterday), they never showed up.

Another issue is the deprecation of QLinkedList in Qt, I replaced it by std::list like the warning suggested, and Undo still seems to work, but really I don't know anything about lists, undo stacks and pop/push model so that's just a shot in the dark.

While testing the program, especially when using undo after some edit operation, I sometimes experience an audible click and the program going mute/stuck play cursor... I let rezound find the backend by itself (can be ALSA, JACK or Pulseaudio) or run it via pipewire's pw-jack, I haven't investigated if any of those make a difference, is that probably the sort of race condition you mention?

As i said, I'm not a programmer, I taught myself a bit C long ago, can read simple C++ code but that's about it.

I leave it there for the moment, cheers!
